test/suites/nodeclaim/nodeclaim_test.go

```diff
@@ -0,0 +1,154 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeclaim_test
+
+import (
+	"github.com/Azure/karpenter/pkg/apis/v1alpha2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1beta1 "github.com/aws/karpenter-core/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter-core/pkg/utils/resources"
+)
+
+var _ = Describe("StandaloneNodeClaim", func() {
+	It("should create a standard NodeClaim within the 'D' sku family", func() {
+		nodeClaim := test.NodeClaim(corev1beta1.NodeClaim{
+			Spec: corev1beta1.NodeClaimSpec{
+				Requirements: []v1.NodeSelectorRequirement{
+					{
+						Key:      v1alpha2.LabelSKUFamily,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"D"},
+					},
+					{
+						Key:      corev1beta1.CapacityTypeLabelKey,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{corev1beta1.CapacityTypeOnDemand},
+					},
+				},
+				NodeClassRef: &corev1beta1.NodeClassReference{
+					Name: nodeClass.Name,
+				},
+			},
+		})
+		env.ExpectCreated(nodeClass, nodeClaim)
+		node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
+		nodeClaim = env.EventuallyExpectCreatedNodeClaimCount("==", 1)[0]
+		Expect(node.Labels).To(HaveKeyWithValue(v1alpha2.LabelSKUFamily, "D"))
+		env.EventuallyExpectNodeClaimsReady(nodeClaim)
+	})
+	It("should create a standard NodeClaim based on resource requests", func() {
+		nodeClaim := test.NodeClaim(corev1beta1.NodeClaim{
+			Spec: corev1beta1.NodeClaimSpec{
+				Resources: corev1beta1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("1"),
```

Comment:
Feel I should know this, but ResourceCPU is a minimum requirement here for provisioning, not a matching requirement?

Reply:
Yes, it's minimum. We're using `Requests` here:
```
// ResourceRequirements describes the compute resource requirements.
type ResourceRequirements struct {
	// Limits describes the maximum amount of compute resources allowed.
	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
	// +optional
	Limits ResourceList `json:"limits,omitempty" protobuf:"bytes,1,rep,name=limits,casttype=ResourceList,castkey=ResourceName"`
	// Requests describes the minimum amount of compute resources required.
	// If Requests is omitted for a container, it defaults to Limits if that is explicitly specified,
	// otherwise to an implementation-defined value. Requests cannot exceed Limits.
	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
	// +optional
	Requests ResourceList `json:"requests,omitempty" protobuf:"bytes,2,rep,name=requests,casttype=ResourceList,castkey=ResourceName"`
	// Claims lists the names of resources, defined in spec.resourceClaims,
	// that are used by this container.
	//
	// This is an alpha field and requires enabling the
	// DynamicResourceAllocation feature gate.
	//
	// This field is immutable. It can only be set for containers.
	//
	// +listType=map
	// +listMapKey=name
	// +featureGate=DynamicResourceAllocation
	// +optional
	Claims []ResourceClaim `json:"claims,omitempty" protobuf:"bytes,3,opt,name=claims"`
}
```

Reply:
Great